### ISSUE TITLE
Add skip

### ISSUE THREAD

**Issue by c0r73x** [AckslD/nvim-trevJ.lua#8]
Hi, could we add the option to skip types?

Like this,

```patch
diff --git a/lua/trevj.lua b/lua/trevj.lua
index 285d57e..64a939d 100644
--- a/lua/trevj.lua
+++ b/lua/trevj.lua
@@ -4,6 +4,7 @@ local ts_utils = require "nvim-treesitter.ts_utils"
 
 local make_default_opts = function()
   return {
+      skip = {},
       final_separator = ',',
       final_end_line = true,
   }
@@ -11,6 +12,7 @@ end
 
 local make_no_final_sep_opts = function()
   return {
+      skip = {},
       final_separator = false,
       final_end_line = true,
   }
@@ -129,6 +131,14 @@ M.format_at_cursor = function()
     end
     for i, child in ipairs(children) do
       local lines = ts_utils.get_node_text(child)
+      if opts.skip then
+        for _, s in ipairs(opts.skip) do
+          if child:type() == s then
+            new_lines[#new_lines] = new_lines[#new_lines] .. vim.fn.join(lines)
+            goto next
+          end
+        end
+      end
       if opts.final_separator and i == #children - 1 then
         if not lines_end_with(lines, opts.final_separator) then
           lines[#lines] = lines[#lines] .. opts.final_separator
@@ -146,6 +156,7 @@ M.format_at_cursor = function()
           new_lines[#new_lines] = new_lines[#new_lines] .. lines[1]
         end
       end
+      ::next::
     end
     vim.api.nvim_buf_set_text(0, srow, scol, erow, ecol, new_lines)
   else
```

To be able to format template tags in vue or html tags. For example.

From:
```vue
<template>
    <h1 foo="bar" :bar="foo" barfoo="x"></h1>
</template>
```

To:
```vue
<template>
    <h1
        foo="bar"
        :bar="foo"
        barfoo="x"
    ></h1>
</template>
```

With:
```lua
vue = {
  start_tag = {
    final_separator = false,
    final_end_line = true,
    skip = { "tag_name" }
  },
}
```

**Comment by AckslD** [comment]
Hi @c0r73x! Great idea :) I added this in #10 with a slightly different implementation. Let me know if that indeed does what you suggested.

**Comment by c0r73x** [comment]
Looks perfect! 👍

**Comment by AckslD** [comment]
Great! :) Do you know if there are other html containers we should add?

**Comment by c0r73x** [comment]
the only other one i can think of is self_closing_tag 🤔

**Comment by AckslD** [comment]
Looking at the [parser](https://github.com/tree-sitter/tree-sitter-html/blob/master/grammar.js) for html it looks like only `start_tag` and `element` uses `repeat`. I guess `element` does not make sense? Can also be added later if needed of course.

**Comment by c0r73x** [comment]
I think you forgot to add the empty skip to the make_default_opts. 

[trevJ] Warning: unsupported option `skip`

After adding it to make_default_opts it works perfectly 👍

**Comment by AckslD** [comment]
Woops indeed, thanks, fixed in #12

Just fyi, I also added the `html` config to the defaults
